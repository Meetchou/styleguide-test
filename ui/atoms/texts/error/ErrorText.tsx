import React, { FC } from 'react';
import cls from 'classnames';
import styled from 'styled-components';

import { ErrorIcon } from '@ui/atoms/icons/icons';

const Error = styled.div`
  svg {
    width: 15px;
    height: 15px;

    margin-top: 1px;
  }
`;

export type ErrorTextProps = {
  className?: string;
  children?: React.ReactNode;
};

const ErrorText:FC<ErrorTextProps> = ({ className, children, ...rest }) => (
  <Error
    {...rest}
    className={cls('flex text-xs text-danger-500', className)}
  >
    <ErrorIcon className='flex-shrink-0 mr-1 text-current' />
    {children}
  </Error>
);

export default ErrorText;
