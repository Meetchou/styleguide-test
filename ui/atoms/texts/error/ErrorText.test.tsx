import React from 'react';

import render from '@ui/testing/render';

import ErrorText from './ErrorText';

describe('atoms/texts/error/ErrorText', () => {
  it('renders correctly', () => {
    const tree = render(<ErrorText>this is a test</ErrorText>);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
