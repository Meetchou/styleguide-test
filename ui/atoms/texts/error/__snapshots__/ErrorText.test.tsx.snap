// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/texts/error/ErrorText renders correctly 1`] = `
.c0 svg {
  width: 15px;
  height: 15px;
  margin-top: 1px;
}

<div
  className="c0 flex text-xs text-danger-500"
>
  <svg
    className="flex-shrink-0 mr-1 text-current"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
    />
  </svg>
  this is a test
</div>
`;
